apps/api/src/services/agent-metrics-helper.ts

```diff
@@ -0,0 +1,214 @@
+import type {
+  AgentMetricsData,
+  AgentPublic,
+  AgentWithMetrics,
+  RawAgentMetricsQueryResult,
+} from "@/types/agent-metrics.js";
+import type { AgentMetadata, AgentStats, AgentTrophy } from "@/types/index.js";
+
+/**
+ * Helper class for transforming and processing agent metrics
+ * Encapsulates business logic for metrics calculations and transformations
+ *
+ * This class handles the conversion between different representations of agent metrics:
+ * 1. RawAgentMetricsQueryResult - Raw database query results
+ * 2. AgentMetricsData - Internal service layer representation (uses null for missing values)
+ * 3. AgentStats - Public API schema (uses undefined for optional fields)
+ *
+ * The separation allows us to follow database conventions internally while
+ * providing clean, minimal JSON responses to API clients.
+ */
+export class AgentMetricsHelper {
+  /**
+   * Transform raw database query results into structured agent metrics data
+   * @param agentIds List of agent IDs to process
+   * @param rawResults Raw query results from repository layer
+   * @returns Array of transformed agent metrics
+   */
+  static transformRawMetricsToAgentMetrics(
+    agentIds: string[],
+    rawResults: RawAgentMetricsQueryResult,
+  ): AgentMetricsData[] {
+    // Create lookup maps for efficient access
+    const agentRanksMap = new Map(
+      rawResults.agentRanks.map((row) => [row.agentId, row]),
+    );
+
+    const competitionCountsMap = new Map(
+      rawResults.competitionCounts.map((row) => [
+        row.agentId,
+        row.completedCompetitions,
+      ]),
+    );
+
+    const voteCountsMap = new Map(
+      rawResults.voteCounts.map((row) => [row.agentId, row.totalVotes]),
+    );
+
+    const tradeCountsMap = new Map(
+      rawResults.tradeCounts.map((row) => [row.agentId, row.totalTrades]),
+    );
+
+    const bestPlacementMap = new Map(
+      rawResults.bestPlacements.map((row) => [
+        row.agentId,
+        {
+          competitionId: row.competitionId,
+          rank: row.rank,
+          score: row.score,
+          totalAgents: row.totalAgents,
+        },
+      ]),
+    );
+
+    const bestPnlMap = new Map(
+      rawResults.bestPnls.map((row) => [
+        row.agentId,
+        {
+          competitionId: row.competitionId,
+          pnl: row.pnl,
+        },
+      ]),
+    );
+
+    // Calculate global ranks from all agent scores
+    const globalRanksMap = new Map<string, number>();
+
+    // Sort all agents by ordinal (descending) to determine ranks
+    const sortedAgents = [...rawResults.allAgentScores].sort(
```

Comment:
Related to the above question: 
With our current prod data the length of `allAgentScores` is ~44, but potentially it will get very large right?  Maybe we need to handle this differently?